frontend/appflowy_flutter/packages/flowy_svg/lib/src/flowy_svg.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+/// The abstract class for FlowySvgData that the code generator will implement
+class FlowySvgData {
+  /// The svg data
+  const FlowySvgData(
+    this.path,
+  );
+
+  /// The path to the svg data in appflowy assets/images
+  final String path;
+}
```

Comment:
Similar to the `IconData` class. This way we can construct a `FlowySvgData` from the path, or by using the constants generated by the script. For example
```dart
FlowySvgData('assets/images/board/check.svg');
-or-
FlowySvgs.check_board;
```